src/test/java/it/gov/pagopa/payment/service/QRCodePaymentServiceImplTest.java

```diff
@@ -0,0 +1,84 @@
+package it.gov.pagopa.payment.service;
+
+import it.gov.pagopa.payment.dto.mapper.TransactionInProgress2SyncTrxStatus;
+import it.gov.pagopa.payment.dto.qrcode.SyncTrxStatusDTO;
+import it.gov.pagopa.payment.enums.SyncTrxStatus;
+import it.gov.pagopa.payment.exception.ClientExceptionNoBody;
+import it.gov.pagopa.payment.model.TransactionInProgress;
+import it.gov.pagopa.payment.repository.TransactionInProgressRepository;
+import it.gov.pagopa.payment.service.qrcode.QRCodeAuthPaymentService;
+import it.gov.pagopa.payment.service.qrcode.QRCodeConfirmationService;
+import it.gov.pagopa.payment.service.qrcode.QRCodeCreationService;
+import it.gov.pagopa.payment.service.qrcode.QRCodePreAuthService;
+import it.gov.pagopa.payment.test.fakers.SyncTrxStatusFaker;
+import it.gov.pagopa.payment.test.fakers.TransactionInProgressFaker;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import java.util.Optional;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class QRCodePaymentServiceImplTest {
+    @Mock
+    TransactionInProgressRepository transactionInProgressRepository;
+    @Mock
+    QRCodeCreationService qrCodeCreationService;
+    @Mock
+     QRCodePreAuthService qrCodePreAuthService;
+    @Mock
+    QRCodeAuthPaymentService qrCodeAuthPaymentService;
+    @Mock
+    QRCodeConfirmationService qrCodeConfirmationService;
+    @Mock
+    TransactionInProgress2SyncTrxStatus transactionMapper;
+    QRCodePaymentServiceImpl qrCodePaymentService;
```

Comment:
evenif it's a test, let's always respect the principle of least privileges

let's use suffix "Mock" to explicitly let know who would read it in the rest of the test, that these variables are mocks

let's separate using empty lines the variables by their "type" (mocks, utilities, tested service), this way we will simplify file reading

```suggestion
    private TransactionInProgressRepository transactionInProgressRepositoryMock;
    @Mock
    private QRCodeCreationService qrCodeCreationServiceMock;
    @Mock
     private QRCodePreAuthService qrCodePreAuthServiceMock;
    @Mock
    private QRCodeAuthPaymentService qrCodeAuthPaymentServiceMock;
    @Mock
    private QRCodeConfirmationService qrCodeConfirmationServiceMock;
    
    private final TransactionInProgress2SyncTrxStatus transactionMapper = new ....;
    
    private QRCodePaymentServiceImpl qrCodePaymentServiceMock;
```